src/index.ts

```diff
@@ -66,6 +66,19 @@ export class Collection<K, V> extends Map<K, V> {
 		return super.clear();
 	}
 
+	/**
+	 * Gets an element if the key exists, otherwise sets and returns {@param defaultValue}.
+	 * @param {*} key - Key to get from/set to the collection.
+	 * @param {*} defaultValue - Default value to be set and returned if the key doesn't exist.
+	 * @returns {*}
+	 */
+	public ensure(key: K, defaultValue: V): V {
+		const value = this.get(key);
+		if (typeof value !== 'undefined') return value;
```

Comment:
This is incorrect if V extends undefined.

Reply:
I did some testing, `typeof null` is an Object and `typeof NaN` is a Number, so unless they manually set undefined as a value I don't think we have to worry about that.

Reply:
Yes, that is why this is incorrect.